subsys/bluetooth/host/scan.c

```diff
@@ -874,7 +880,7 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 			 * this is the first report from the new advertiser.
 			 * Initialize the new advertiser.
 			 */
-			__ASSERT_NO_MSG(reassembling_advertiser.state == FRAG_ADV_INACTIVE);
+			__ASSERT_NO_MSG(reassembling_advertiser.state != FRAG_ADV_REASSEMBLING);
```

Comment:
This seems like a side effect of not going to discarding state.

Reply:
Don't think so, if set `FRAG_ADV_DISCARDING`and return, and then receive new advertising, will enter here:
```C
		if (is_new_advertiser) {
			/* We are not reassembling reports from an advertiser and
			 * this is the first report from the new advertiser.
			 * Initialize the new advertiser.
			 */
			__ASSERT_NO_MSG(reassembling_advertiser.state == FRAG_ADV_INACTIVE);
			init_reassembling_advertiser(&evt->addr, evt->sid);
		}

```

Reply:
@jhedberg Ping.